{
	"schema_version": "1.6.0",
	"id": "UBUNTU-CVE-2021-42343",
	"modified": "2025-01-13T10:22:37Z",
	"published": "2021-10-26T11:15:00Z",
	"related": [
		"CVE-2021-42343"
	],
	"details": "An issue was discovered in the Dask distributed package before 2021.10.0 for Python. Single machine Dask clusters started with dask.distributed.LocalCluster or dask.distributed.Client (which defaults to using LocalCluster) would mistakenly configure their respective Dask workers to listen on external interfaces (typically with a randomly selected high port) rather than only on localhost. A Dask cluster created using this method (when running on a machine that has an applicable port exposed) could be used by a sophisticated attacker to achieve remote code execution.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Ubuntu:Pro:18.04:LTS",
				"name": "dask.distributed",
				"purl": "pkg:deb/ubuntu/dask.distributed@1.20.2+ds.1-2?arch=source&distro=esm-apps/bionic"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"1.18.0+ds.1-1",
				"1.20.2+ds.1-1",
				"1.20.2+ds.1-2"
			],
			"ecosystem_specific": {
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2021/UBUNTU-CVE-2021-42343.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:20.04:LTS",
				"name": "dask.distributed",
				"purl": "pkg:deb/ubuntu/dask.distributed@1.25.0+ds.1-1?arch=source&distro=focal"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"1.25.0+ds.1-1"
			],
			"ecosystem_specific": {
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2021/UBUNTU-CVE-2021-42343.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:22.04:LTS",
				"name": "dask.distributed",
				"purl": "pkg:deb/ubuntu/dask.distributed@2022.01.0+ds.1-1?arch=source&distro=jammy"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2022.01.0+ds.1-1"
						}
					]
				}
			],
			"versions": [
				"2021.01.0+ds.1-2.1",
				"2021.09.1+ds.1-3"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-distributed-doc",
						"binary_version": "2022.01.0+ds.1-1"
					},
					{
						"binary_name": "python3-distributed",
						"binary_version": "2022.01.0+ds.1-1"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2021/UBUNTU-CVE-2021-42343.json"
			}
		}
	],
	"references": [
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/CVE-2021-42343"
		},
		{
			"type": "REPORT",
			"url": "https://www.cve.org/CVERecord?id=CVE-2021-42343"
		}
	]
}
